assessor/lib/locations.rb

```diff
@@ -0,0 +1,99 @@
+
+# ......................................................................
+# Engine-specific constructed values
+
+def testcase_of(engine)
+  return "_minibroker_postgres" if engine =~ /postgresql/
+  "_minibroker_#{engine}"
+end
+
+def patch_of(engine)
+  File.join(@top, "assessor/patches", "#{engine}.patch")
+end
+
+# ......................................................................
+# Various (semi)constant values, mostly paths and the like
+
+def mbbt_repository
+  "https://minibroker-helm-charts.s3.amazonaws.com/kubernetes-charts"
+end
+
+def fixed_time
+  "2018-07-30T17:55:01.330815339Z"
+end
+
+def stable
+  "https://kubernetes-charts.storage.googleapis.com"
+end
+
+def helm_app
+  "chart-under-test"
+end
+
+def chart_in_app
+  "chart.tgz"
+end
+
+def statepath
+  @statepath ||= File.join(@workdir, 'results.yaml')
+end
+
+def appdir
+  @appdir ||= File.join(@workdir, "charts")
+end
+
+def working_saved
+  @wsaved ||= File.join(archive_saved, 'index.yaml')
+end
+
+def archive_saved
+  @asaved ||= File.join(@workdir, 'ok')
+end
+
+def archive_orig
+  @aunpatched ||= File.join(@workdir, 'archive-orig.tgz')
+end
+
+def archive_patched
+  @apatched ||= File.join(@workdir, 'archive-patched.tgz')
+end
+
+def manifest
+  @manifest ||= File.join(appdir, "manifest.yml")
+end
+
+def chart_index
+  @chartindex ||= File.join(appdir, "index.yaml")
+end
+
+def archive_app
+  @aapp ||= File.join(appdir, chart_in_app)
+end
+
+def domain
+  unless @domain
+    pod, _, _ = capture "kubectl", "get", "pods", "-o", "yaml", "--namespace", @namespace, "api-group-0"
+    @domain = YAML.load(pod)['spec']['containers'][0]['env'].select { |ev| ev['name'] == "DOMAIN" }.first['value']
```

Comment:
Prefer configmap if possible.

Reply:
Configmap has no domain information:
```
vagrant@vagrant-kube:~/a> k get configmap cf:secrets-config -o yaml
fetching configmap => 6
kubectl get configmap secrets-config --namespace cf -o yaml
apiVersion: v1
data:
  config-version: "1"
  current-secrets-generation: "1"
  current-secrets-name: secrets-2.17.1-1
  previous-secrets-name: ""
kind: ConfigMap
metadata:
  creationTimestamp: "2019-08-14T17:29:59Z"
  name: secrets-config
  namespace: cf
  resourceVersion: "2859"
  selfLink: /api/v1/namespaces/cf/configmaps/secrets-config
  uid: 242c9672-beb9-11e9-a01e-5254004e2efc
```

Neither has the fixed secret:
```
vagrant@vagrant-kube:~/a> k get secrets cf:secrets$ -o yaml|grep -i domain
fetching secrets => 17
kubectl get secrets secrets --namespace cf -o yaml
```

Nor the versioned secret
```
vagrant@vagrant-kube:~/a> k get secrets cf:secrets- -o yaml|grep -i domain
fetching secrets => 17
kubectl get secrets secrets-2.17.1-1 --namespace cf -o yaml
```

Reply:
Ah ☹️  It's in helm (`helm get values cf`), but that requires a different tool.

I guess what we have is good enough then; the best alternative I have is
```bash
kubectl get -n cf statefulset/api-group -o jsonpath='{.spec.template.spec.containers[0].env[?(@.name=="DOMAIN")].value}'
```
But honestly I can't claim that as _better_.